### DISCUSSION TITLE
Can the command line do thumbnails and upload to printer?

### DISCUSSION THREAD

**Discussion by cs2dsb** [supermerill/SuperSlicer#1994] [Q&A]
Hi,

I've been trying to use the command line to generate large lithophane prints as the gui can freeze or crash if the lithophanes are too detailed but there are two features I can't seem to get the command line to do:

1. Embedding thumbnails in the gcode
2. Uploading to the printer over http

The final gcode has `; thumbnails = 48x48,300x300` so I can see it's got the right config for that in theory but the file doesn't contain any thumbnails.

I can also see the --host-type flag in the command line help but nothing to tell it to use the remote host. I'm passing in the exact same 4 config files the gui is using (physical_printer, printer, print  and filament) via the `--load` flag so it should have the right host settings.

Thanks for any help!

The exact command I'm using is:
```
./SuperSlicer-ubuntu_18.04-2.3.56.9.AppImage \
--slice \
--load ~/.config/SuperSlicer/physical_printer/Abomination.ini \
--load ~/.config/SuperSlicer/printer/Creality\ Ender-3\ V2\ -\ mine.ini \
--load ~/.config/SuperSlicer/print/0.20mm\ LithoPhane.ini \
--load ~/.config/SuperSlicer/filament/Generic\ PLA\ @CREALITY\ -\ Copy.ini \
--host-type klipper \
~/Downloads/lithophanes/jake/ItsLitho_DSC00850.stl
```

**Comment by supermerill** [comment]
> Uploading to the printer over http

I don't think it's supported in the command line. I think the rational is that you may be able to do this with a script in command line anyway.

For the thumbnail, I'll look why.
#2000

**Reply by cs2dsb** [reply]
Ah, I just thought it would be nice so the host and stuff is all managed in the physical printer config but as you say, I can just do it with a script. Could probably even pull the details out of the superslicer config files without too much hassle.

Thanks for looking at the thumbnails.

**Reply by cs2dsb** [reply]
I've been doing a bit more experimenting with it and found a few other issues where it doesn't produce the same results as the GUI. Is it worth me raising issues about each thing? I don't want to make a load of work for you if no one is using the command line.

Some examples of different behaviour: 
1.`M204 S500` at the top of the command line file, not in the gui one. `; machine_limits_usage = time_estimate_only` at the bottom of both. I'm using klipper so the command line version ended up more than doubling the real print time
2. `; perimeters extrusion width = 0.45mm` in the command line file, `; perimeters extrusion width = 0.46mm` in the gui one. All relevant settings appear to be the same in the bottom of the file.

For both of the above I exported gcode from the gui then exported the config to a single ini file to make sure the right settings were being passed to the command line

**Reply by supermerill** [reply]
> For both of the above I exported gcode from the gui then exported the config to a single ini file to make sure the right settings were being passed to the command line

you can also export the project file and use only that (no ini) in the command line.

Will look.

**Reply by supermerill** [reply]
I'm stupid, of course there isn't a thumbnail: there is no GUI to take a screenshot, that's the whole point of the command line!

**Reply by cs2dsb** [reply]
Ah, I didn't realize it was taking a screenshot of the GUI for the thumbnail. I found a few command line based stl thumbnail tools so I can try one of those after generating the gcode.

Thanks for looking.
